go/merkletree/merkle_verifier.go

```diff
@@ -0,0 +1,176 @@
+package merkletree
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+)
+
+// MerkleVerifier is a class which knows how to verify merkle inclusion and consistency proofs.
```

Comment:
Yeah - "audit proof" is kinda vague, so we settled on using "inclusion-" and "consistency-proof" for clarity.